contracts/axelar-gateway/src/test_gateway.rs

```diff
@@ -476,3 +489,65 @@ fn transfer_operatorship_unauthorized() {
         }])
         .transfer_operatorship(&new_operator);
 }
+
+#[test]
+fn transfer_ownership() {
+    let (env, contract_id, client) = setup_env();
+    let owner = Address::generate(&env);
+    let operator = Address::generate(&env);
+    let new_owner = Address::generate(&env);
+
+    initialize(&env, &client, owner.clone(), operator, 1, randint(1, 10));
+
+    assert_eq!(client.owner(), owner);
+
+    client
+        .mock_auths(&[MockAuth {
+            address: &owner,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "transfer_ownership",
+                args: (&new_owner,).into_val(&env),
+                sub_invokes: &[],
+            },
+        }])
+        .transfer_ownership(&new_owner);
+
+    assert_last_emitted_event(
+        &env,
+        &contract_id,
+        (
+            String::from_str(&env, "ownership_transferred"),
+            owner,
+            new_owner.clone(),
+        ),
+        (),
+    );
+
+    assert_eq!(client.owner(), new_owner);
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Auth, InvalidAction)")] // Unauthorized
```

Comment:
use `try_*` version of the function and assert err using the new pattern from Laith

Reply:
I attempted to add `GatewayError::Unauthorized` and integrate `ensure!` logic for `try_` error handling. However, since `require_auth()` panics, it can't be handled by the `try_*` functions. I confirmed this with Laith. @milapsheth Let me know your thoughts.

Reply:
I see, define the error string as a constant so we can reuse it for all auth failure tests. I'm a bit surprised that it panics vs returns a certain type of error. Ask stellar team about it so we know how to do this in the future?

Reply:
I tried to define `const ERROR_STRING: &str = "HostError: Error(Auth, InvalidAction)";` and tried to use it for all auth failure test, it won't let me use a variable to the `expected=` section. I created [a ticket](https://axelarnetwork.atlassian.net/browse/AXE-6136) to track this issue, will ask to the stellar team.